lib/cli/ui/prompt.rb

```diff
@@ -48,26 +49,29 @@ class << self
         # Free form question when the answer can be empty
         #   CLI::UI::Prompt.ask('What is your opinion on this question?', allow_empty: true)
         #
-        # Question with answers
-        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python))
+        # Multiple choice question with defined handlers
+        #   CLI::UI::Prompt.ask('What kind of project is this?', options: %w(rails go ruby python)) do |handler|
+        #     handler.add_option('rails')  { |selection| puts selection } => outputs "rails" if selected
+        #     handler.add_option('go')     { |selection| puts selection } => outputs "go" if selected
+        #     handler.add_option('ruby')   { |selection| puts selection } => outputs "ruby" if selected
+        #     handler.add_option('python') { |selection| puts selection } => outputs "python" if selected
         #
         #
-        def ask(question, options: nil, default: nil, is_file: nil, allow_empty: true)
-          if (default && !allow_empty) || (options && (default || is_file))
+        def ask(question, default: nil, is_file: nil, allow_empty: true, &define_options)
+          if (default && !allow_empty) || (block_given? && (default || is_file))
             raise(ArgumentError, 'conflicting arguments')
           end
 
-          if default
-            puts_question("#{question} (empty = #{default})")
-          elsif options
-            puts_question("#{question} {{yellow:(choose with ↑ ↓ ⏎)}}")
-          else
-            puts_question(question)
-          end
-
           # Present the user with options
-          if options
-            resp = InteractiveOptions.call(options)
+          if block_given?
+            require 'cli/ui/prompt/options_handler'
+            handler = OptionsHandler.new
+
+            yield handler
+            raise(ArgumentError, 'insufficient options') if handler.options.size < 2
```

Comment:
Why is 1 option insufficient?

Reply:
Multiple choice from 1 option? Even then you probably want a yes/no confirm.